compiler/rustc_smir/src/stable_mir/compiler_interface.rs

```diff
@@ -22,47 +23,116 @@ use stable_mir::{
     ItemKind, Symbol, TraitDecls, mir,
 };
 
-use crate::stable_mir;
+use crate::{rustc_smir, stable_mir};
+
+/// Stable public API for querying compiler information.
+///
+/// All queries are delegated to an internal [`Context`] that provides
+/// similar APIs but based on internal rustc constructs.
+///
+/// Do not use this directly. This is currently used in the macro expansion.
+pub struct SmirInterface<'tcx> {
```

Comment:
On the other hand, this one should be `pub(crate)`, since users of `stable_mir` should not interact with it.